controllers/discovery/initDiscoveryUserData.js

```diff
@@ -59,7 +110,16 @@ const InitDiscoveryUserData = async (req, res) => {
             A.is_anonymous,
             A.combined_user_score,
             A.karma_score,
+            A.allow_anon_dm,
+            A.only_received_dm_from_user_following,
             ${similarTopicQuery} as community_info,
+            (SELECT COUNT(*) FROM user_follow_user WHERE user_id_followed = A.user_id) AS followersCount,
+            EXISTS (
+                SELECT 1
+                FROM user_follow_user AS f2
+                WHERE f2.user_id_follower = :userId
+                AND f2.user_id_followed = A.user_id
+            ) AS is_followed,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The subquery for checking if a user is followed can be optimized by adding a LIMIT clause. This will stop the query as soon as it finds a match, improving performance.
```diff
- EXISTS (
-     SELECT 1
-     FROM user_follow_user AS f2
-     WHERE f2.user_id_follower = :userId
-     AND f2.user_id_followed = A.user_id
- ) AS is_followed,
+ EXISTS (
+     SELECT 1
+     FROM user_follow_user AS f2
+     WHERE f2.user_id_follower = :userId
+     AND f2.user_id_followed = A.user_id
+     LIMIT 1
+ ) AS is_followed,
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->